### ISSUE TITLE
Leave `Activity.Status` unset when `Dispose()` is called without `Complete()`

### ISSUE THREAD

**Issue by srogovtsev** [serilog-tracing/serilog-tracing#42]
TL/DR: _SerilogTracing_ never outputs activities with the status of `Unset`, while this is the default behavior for `System.Diagnostics.Activity`.

The code is (I've tried to choose the closest choices in case of _SerilogTracing_; there's no `AddException` or something):

```cs
source.StartActivity("Activity - Regular").Dispose();

using (var a = source.StartActivity("Activity - Completed"))
    a?.SetStatus(ActivityStatusCode.Ok);

using (var a = source.StartActivity("Activity - Exception"))
    a?.RecordException(new Exception());

using (var a = source.StartActivity("Activity - Error"))
    a?.SetStatus(ActivityStatusCode.Error);

(logger.StartActivity("Serilog - Regular")).Dispose();

using (var a = logger.StartActivity("Serilog - Completed"))
    a.Complete();

using (var a = logger.StartActivity("Serilog - Exception"))
    a.Complete(exception: new Exception());

using (var a = logger.StartActivity("Activity - Error"))
    a.Complete(LogEventLevel.Error);
```

Output:

```
Program:Activity - Regular stopped with status Unset
Program:Activity - Completed stopped with status Ok
Program:Activity - Exception stopped with status Unset
Program:Activity - Error stopped with status Error
Serilog:Serilog - Regular stopped with status Ok <--
Serilog:Serilog - Completed stopped with status Ok
Serilog:Serilog - Exception stopped with status Ok <--
Serilog:Activity - Error stopped with status Error
```

According to [the documentation](https://opentelemetry.io/docs/concepts/signals/traces/#span-status), spans can have _three_ statuses: `Unset`, `Error` and `Ok`:

> To reiterate: Unset represents a span that completed without an error. Ok represents when a developer explicitly marks a span as successful. In most cases, it is not necessary to explicitly mark a span as Ok.

`System.Diagnostics.Activity` is consistent with this, and doesn't set activity status unless explicitly asked to do so. This is what it looks like:

<details><summary>Zipkin</summary>

Unset:
![image](https://github.com/serilog-tracing/serilog-tracing/assets/2332517/a938b29f-544d-40b7-a5b4-69ef0b5589ba)

Ok:
![image](https://github.com/serilog-tracing/serilog-tracing/assets/2332517/06800c2b-0e50-4e50-a31b-0f409b16582f)
</details> 

<details><summary>Console (standard otel console exporter)</summary>

Unset:
```
Activity.TraceId:            760ca0cf2b9d8250091827066457cd14
Activity.SpanId:             a4b39aa8914d4567            
Activity.TraceFlags:         Recorded                    
Activity.ParentSpanId:       296d7f41b175cf2e            
Activity.ActivitySourceName: Program                     
Activity.DisplayName:        Activity - Regular          
Activity.Kind:               Internal                    
Activity.StartTime:          2024-01-24T19:07:26.7426619Z
Activity.Duration:           00:00:00.0005463                
Resource associated with Activity:                           
    telemetry.sdk.name: opentelemetry                        
    telemetry.sdk.language: dotnet                           
    telemetry.sdk.version: 1.7.0                             
    service.name: unknown_service:ActivityTracing            
```

Ok:
```                                                             
Activity.TraceId:            760ca0cf2b9d8250091827066457cd14
Activity.SpanId:             fd5ef238abf57ff2                
Activity.TraceFlags:         Recorded                        
Activity.ParentSpanId:       296d7f41b175cf2e                
Activity.ActivitySourceName: Program                         
Activity.DisplayName:        Activity - Completed            
Activity.Kind:               Internal                        
Activity.StartTime:          2024-01-24T19:07:26.7913852Z    
Activity.Duration:           00:00:00.0000639                
StatusCode: Ok
Resource associated with Activity:
    telemetry.sdk.name: opentelemetry
    telemetry.sdk.language: dotnet
    telemetry.sdk.version: 1.7.0
    service.name: unknown_service:ActivityTracing
```
</details>

While I understand where this probably comes from (_SerilogTimings_ auto-completes operations by default), this is much less useful, as it's not consistent with other distributed tracing implementations. And it's probably very easy to fix as well.

PS This might even tie-in nicely with https://github.com/serilog-tracing/serilog-tracing/issues/18#issuecomment-1902111977, although it was wrong of me to expect `aborted` - otel terminology is different here.

**Comment by srogovtsev** [comment]
One more point in case:

> [Span Status](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/trace/api.md#set-status) MUST be left unset if HTTP status code was in the 1xx, 2xx or 3xx ranges

(https://opentelemetry.io/docs/specs/semconv/http/http-spans/#status)

**Comment by nblumhardt** [comment]
Hey that's really interesting, thanks for the notes @srogovtsev :+1: 

RE the OpenTelemetry span status requirement, I understand it's probably not your point here, but in your scenario using the OTel instrumentation and not enabling the SerilogTracing instrumentation would ensure you get this behavior as expected.

I think purely from SerilogTracing's point of view, the "unset/abandoned" connection is an interesting one to make, and it does provide additional information about the span that the current scheme we're using is not capturing.

Modifying `LoggerActivity` to separate `Dispose()` behavior from `Complete()` behavior would achieve what you're suggesting on the `Systems.Diagnostic.Activity` side, I'll think through the implications in the rest of the pipeline and loop back here.

**Comment by srogovtsev** [comment]
> RE the OpenTelemetry span status requirement, I understand it's probably not your point here, but in your scenario using the OTel instrumentation and not enabling the SerilogTracing instrumentation would ensure you get this behavior as expected.

Just to clarify, I wasn't using any instrumentation here, but simply generated the activities/spans by hand.

The HTTP span example was just an example where the spec unequivocally says the status should be unset, so if, for example, one decides to implement their own instrumentation by using `StartActivity` (and frankly, that's exactly what we had done previously with _SerilogTimings_), they will find themselves in a position where the generated activity is non-compliant.

> I think purely from SerilogTracing's point of view, the "unset/abandoned" connection is an interesting one to make

I'd like to point out - because it's the mistake that I myself have made recently - that "unset" in OpenTelemetry isn't "abandoned": to the contrary, it is the default "all is as expected" status (and there's no "abandoned" in OpenTelemetry, just error and _maybe_ description to accompany it).

> Modifying LoggerActivity to separate Dispose() behavior from Complete() behavior would achieve what you're suggesting on the Systems.Diagnostic.Activity side

I was thinking along these lines myself, just didn't want to impose my opinion on this design.

**Comment by nblumhardt** [comment]
From the `Activity` standpoint the status values have slightly different meanings, with the mapping to whatever OTel expects being a bit further along in the process (https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activitystatuscode?view=net-8.0) - the key to making things as interoperable as possible might be down in the OTLP mapping. All good to consider :+1:
